engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/TypedMap.scala

```diff
@@ -1,3 +1,63 @@
 package pl.touk.nussknacker.engine.api.typed
 
-case class TypedMap(fields: Map[String, Any])
\ No newline at end of file
+import java.{util => ju}
+
+case class TypedMap(fields: Map[String, Any]) extends ju.AbstractMap[String, Any] { self =>
```

Comment:
I wonder - maybe we should use some map directly? Or is it for backwards compatibility?

Reply:
Indeed it will be less magic if we keep java Map for fields. Also I think that will be better if we will have equals working correctly with other maps - because of this I've replaced case class with plain serializable class.

I've added apply method for backward compatibility, and written some notice in MigrationGuide in case if someone use pattern matching on TypedMap